### PR TITLE
xSMD crafting recipes

### PR DESCRIPTION
**Pull request by GTNH-Colen**
These are intended for wetware lines and above.

### CODE REVIEW

**Comment by GTNH-Colen** [comment]
Open to feedback on these but I am personally happy with this as the final iteration after some thoughts and discussion with others.

**Comment by yukieiji** [comment]
Personally, I think it is a good idea to double the number of outputs of the craft and reduce the consumption by a quarter for the following reasons.
1. Tiers that change from SMD to ASMD consume less ASMD
2. xSMD creation difficulty and assembly line must be used.

**Comment by GTNH-Colen** [comment]
> Personally, I think it is a good idea to double the number of outputs of the craft and reduce the consumption by a quarter for the following reasons.
> 
> 1. Tiers that change from SMD to ASMD consume less ASMD
> 2. xSMD creation difficulty and assembly line must be used.

The xSMDs will only be used in wetware plus (effectively late UV/UHV+) and will fully replace the current ASMD requirements there. The assline is used but only very sparingly, 1 LuV pump can make 960 xSMD diodes and only 1 of the SMDs utilise this fact. The others have IV SCs. It's 1 pump per 15 LuV SC and 1 LuV SC per. This seems fair for the tier it is at imo.

**Comment by yukieiji** [comment]
What tier of circuit crafting does xSMD support?(I'm thinking that with this consumption, I'll need another higher level SMD material before MaxV)
Current SMD consumption is as follows. I don't have to tell you that the recipes need to be balanced, but they also need to be adjusted according to the current SMD consumption.
SMD:Integrated(4 SMD) ～Quantam(64 SMD)
ASMD:Integrated(1 ASMD) ～Bio(64 ASMD)

**Comment by GTNH-Colen** [comment]
It supports wetware, bio and the misc circuits above that. Nothing else. Higher tier SMDs will be introduce as more circuits are added, so worrying about their use down the line is moot for now. I'm not sure I follow the rest of your post.

Bios from my napkin maths use 96 diodes for example, this would mean 1/10th of 1 LuV pump per UEV circuit. That seems fair to me.

**Comment by GlodBlock** [comment]
do you just choose some random expensive material for xsmd? honestly, i can't get the point of why choose them

**Comment by GTNH-Colen** [comment]
> do you just choose some random expensive material for xsmd? honestly, i can't get the point of why choose them

No this was deliberately chosen to integrate GT++ materials better into the pack (these materials are all ZPM tier in the ABS). I didn't realise there was a correct xSMD recipe that uses materials you approved of, can you share it with me if so?

**Comment by GlodBlock** [comment]
the problem isn't tier or gt++ but the material itself.
these gt++ alloy are classic heat/pressure/corrosion material for their origin use in gt++, how they will be suitable for these mirco ciruit part?
resistor should choose materials with high resistance and voltage like duranium/naquadah alloy.
transistor and diode material should be semiconductor, but there is not enough suitable choice in game currently, quantum alloy containing germanium and gallium so seems fine but it is uhv.
capacitor material should be able to hold great potential difference like zirconia.
what you choosed should be the machine part material instead of the ciruit.

**Comment by GTNH-Colen** [comment]
Dude it's a game with made up space metals, I'm really not going into super realism of the exact conductivity of each material and how it would hypothetically interact in a made up circuit.

Like I get simple realism such as using copper as a conductor etc but this is extreme endgame with made up alloys/metals, no one is playing GTNH and thinking "damn why is Pikyonium 64B used in a SMD, the chemical composition makes no sense!".

**Comment by boubou19** [comment]
why do i have the feeling that people try to find reasons to reject GT++ alloys because they come from GT++ 🤔

**Comment by GlodBlock** [comment]
i am not mean to follow the reality strictly but at least not completely go beyond it. like why don't volcanus use cryotheum as fuel because it counters common sense. ofc both cryotheum and pytrotheum are made up material from TE.
also most gt++ alloy exist in real life, few from other game like world of warcraft, only very very few is completely made up and even these made up one are following certain rules like the quantum alloy i mentioned above.

**Comment by GTNH-Colen** [comment]
These alloys are literally named after people in the community, they are made up names.

- Pikyonium = Piky
- Lafium = 0lafe
- Cinobite = Cinobi
- Arceus2B = Pokemon I guess? Lol

File: src/main/java/com/dreammaster/bartworksHandler/BacteriaRegistry.java [hunk 1]
```diff
@@ -286,7 +286,7 @@ private void runBWRecipes(){
         addBacterialVatRecipe(
                 new ItemStack[]{AntimonyTrioxide.getDust(16), Osmium.getDust(16)},
                 CultureSet.get("CombinedBac"),
-                new FluidStack[]{Oil.getFluid(1000)},
+                new FluidStack[]{Oil.getFluid(1)},
```
Comment (Dream-Master) [inline src/main/java/com/dreammaster/bartworksHandler/BacteriaRegistry.java:289]: this is change we need to discuss. You cahnage from 1000 to 1.
And this change not belongs to the xsmd's so please seperate it

File: src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java [hunk 2]
```diff
@@ -2713,19 +2717,78 @@ public void run3() {
...
         GT_Values.RA.addAssemblerRecipe(new ItemStack[]{GT_OreDictUnificator.get(OrePrefixes.foil, Materials.VanadiumGallium, 2), GT_OreDictUnificator.get(OrePrefixes.wireFine, Materials.HSSG, 16), GT_OreDictUnificator.get(OrePrefixes.bolt, Materials.TungstenSteel, 4), GT_Utility.getIntegratedCircuit(6)}, Materials.Polybenzimidazole.getMolten(576L), ItemList.Circuit_Parts_TransistorASMD.get(64L), 300, 1920);
         GT_Values.RA.addAssemblerRecipe(new ItemStack[]{GT_OreDictUnificator.get(OrePrefixes.foil, Materials.Polybenzimidazole, 4), GT_OreDictUnificator.get(OrePrefixes.foil, Materials.HSSS, 2), GT_OreDictUnificator.get(OrePrefixes.bolt, Materials.TungstenSteel, 4), GT_Utility.getIntegratedCircuit(6)}, Materials.Polybenzimidazole.getMolten(576L), ItemList.Circuit_Parts_CapacitorASMD.get(64L), 300, 1920);
+        // xSMD
```
Comment (Dream-Master) [inline src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java:2735]: I would like so see one more component in the recipe so we add each tier a component

File: src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java [hunk 3]
```diff
@@ -3381,17 +3444,17 @@ public void run3() {
...
-            GT_Values.RA.addCircuitAssemblerRecipe(new ItemStack[]{ItemList.Circuit_Chip_NeuroCPU.get(1L), ItemList.Circuit_Chip_CrystalCPU.get(1L), ItemList.Circuit_Chip_NanoCPU.get(1L), ItemList.Circuit_Parts_CapacitorASMD.get(8L), ItemList.Circuit_Parts_TransistorASMD.get(8L), GT_OreDictUnificator.get(OrePrefixes.wireFine, Materials.YttriumBariumCuprate, 8)}, tMat.getMolten(144L * tMultiplier / 2L), ItemList.Circuit_Neuroprocessor.get(1L), 200, 38400, true);
+            GT_Values.RA.addCircuitAssemblerRecipe(new ItemStack[]{ItemList.Circuit_Chip_NeuroCPU.get(1L), ItemList.Circuit_Chip_CrystalCPU.get(1L), ItemList.Circuit_Chip_NanoCPU.get(1L), ItemList.Circuit_Parts_CapacitorXSMD.get(8L), ItemList.Circuit_Parts_TransistorXSMD.get(8L), GT_OreDictUnificator.get(OrePrefixes.wireFine, Materials.YttriumBariumCuprate, 8)}, tMat.getMolten(144L * tMultiplier / 2L), ItemList.Circuit_Neuroprocessor.get(1L), 200, 38400, true);
```
Comment (Dream-Master) [inline src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java:3447]: Please not exchange asmd with xsmd. Xsmd are better asmds so they need to be add to make new recipes with lower amount of xsmd's

**Review by Dream-Master — CHANGES_REQUESTED** [review]

File: src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java (hunk 2 above)
Comment (Dream-Master) [reply src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java:2735]: yea looks good now. Will check ingame

File: src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java [hunk 4]
```diff
@@ -3382,16 +3449,20 @@ public void run3() {
...
             GT_Values.RA.addCircuitAssemblerRecipe(new ItemStack[]{ItemList.Circuit_Chip_NeuroCPU.get(1L), ItemList.Circuit_Chip_CrystalCPU.get(1L), ItemList.Circuit_Chip_NanoCPU.get(1L), ItemList.Circuit_Parts_CapacitorASMD.get(8L), ItemList.Circuit_Parts_TransistorASMD.get(8L), GT_OreDictUnificator.get(OrePrefixes.wireFine, Materials.YttriumBariumCuprate, 8)}, tMat.getMolten(144L * tMultiplier / 2L), ItemList.Circuit_Neuroprocessor.get(1L), 200, 38400, true);
+            GT_Values.RA.addCircuitAssemblerRecipe(new ItemStack[]{ItemList.Circuit_Chip_NeuroCPU.get(1L), ItemList.Circuit_Chip_CrystalCPU.get(1L), ItemList.Circuit_Chip_NanoCPU.get(1L), ItemList.Circuit_Parts_CapacitorXSMD.get(2L), ItemList.Circuit_Parts_TransistorXSMD.get(2L), GT_OreDictUnificator.get(OrePrefixes.wireFine, Materials.YttriumBariumCuprate, 8)}, tMat.getMolten(144L * tMultiplier / 2L), ItemList.Circuit_Neuroprocessor.get(1L), 200, 38400, true);
```
Comment (Dream-Master) [inline src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java:3452]: i would suggest tier up this one tier like smd recipes are cheaper than asmd recipe (tier wise)

File: src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java [hunk 5]
```diff
@@ -3382,16 +3449,20 @@ public void run3() {
...
             GT_Values.RA.addCircuitAssemblerRecipe(new ItemStack[]{ItemList.Circuit_Board_Wetware_Extreme.get(1L), ItemList.Circuit_Neuroprocessor.get(2L), ItemList.Circuit_Parts_Coil.get(32L), ItemList.Circuit_Parts_CapacitorASMD.get(12L), ItemList.Circuit_Chip_Ram.get(24L), GT_OreDictUnificator.get(OrePrefixes.wireFine, Materials.YttriumBariumCuprate, 16)}, tMat.getMolten(144L * tMultiplier), ItemList.Circuit_Wetwarecomputer.get(1L), 300, 38400, true);
+            GT_Values.RA.addCircuitAssemblerRecipe(new ItemStack[]{ItemList.Circuit_Board_Wetware_Extreme.get(1L), ItemList.Circuit_Neuroprocessor.get(2L), ItemList.Circuit_Parts_Coil.get(32L), ItemList.Circuit_Parts_CapacitorXSMD.get(3L), ItemList.Circuit_Chip_Ram.get(24L), GT_OreDictUnificator.get(OrePrefixes.wireFine, Materials.YttriumBariumCuprate, 16)}, tMat.getMolten(144L * tMultiplier), ItemList.Circuit_Wetwarecomputer.get(1L), 300, 38400, true);
```
Comment (Dream-Master) [inline src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java:3455]: i would suggest tier up this one tier like smd recipes are cheaper than asmd recipe (tier wise)

File: src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java [hunk 6]
```diff
@@ -3382,16 +3449,20 @@ public void run3() {
...
             GT_Values.RA.addCircuitAssemblerRecipe(new ItemStack[]{ItemList.Circuit_Chip_BioCPU.get(1L), ItemList.Circuit_Chip_CrystalSoC2.get(1L), ItemList.Circuit_Chip_NanoCPU.get(2L), ItemList.Circuit_Parts_CapacitorASMD.get(12L), ItemList.Circuit_Parts_TransistorASMD.get(12L), GT_OreDictUnificator.get(OrePrefixes.wireFine, Materials.NiobiumTitanium, 16)}, tMat.getMolten(144L * tMultiplier / 2L), ItemList.Circuit_Bioprocessor.get(1L), 300, 153600, true);
+            GT_Values.RA.addCircuitAssemblerRecipe(new ItemStack[]{ItemList.Circuit_Chip_BioCPU.get(1L), ItemList.Circuit_Chip_CrystalSoC2.get(1L), ItemList.Circuit_Chip_NanoCPU.get(2L), ItemList.Circuit_Parts_CapacitorXSMD.get(3L), ItemList.Circuit_Parts_TransistorXSMD.get(3L), GT_OreDictUnificator.get(OrePrefixes.wireFine, Materials.NiobiumTitanium, 16)}, tMat.getMolten(144L * tMultiplier / 2L), ItemList.Circuit_Bioprocessor.get(1L), 300, 153600, true);
```
Comment (Dream-Master) [inline src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java:3462]: i would suggest tier up this one tier like smd recipes are cheaper than asmd recipe (tier wise)

File: src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java [hunk 7]
```diff
@@ -3382,16 +3449,20 @@ public void run3() {
...
             GT_Values.RA.addCircuitAssemblerRecipe(new ItemStack[]{ItemList.Circuit_Board_Bio_Ultra.get(1L), ItemList.Circuit_Bioprocessor.get(2L), ItemList.Circuit_Parts_Coil.get(48L), ItemList.Circuit_Parts_CapacitorASMD.get(16L), ItemList.Circuit_Chip_Ram.get(32L), GT_OreDictUnificator.get(OrePrefixes.wireFine, Materials.NiobiumTitanium, 24)}, tMat.getMolten(144L * tMultiplier), ItemList.Circuit_Biowarecomputer.get(1L), 400, 153600, true);
+            GT_Values.RA.addCircuitAssemblerRecipe(new ItemStack[]{ItemList.Circuit_Board_Bio_Ultra.get(1L), ItemList.Circuit_Bioprocessor.get(2L), ItemList.Circuit_Parts_Coil.get(48L), ItemList.Circuit_Parts_CapacitorXSMD.get(4L), ItemList.Circuit_Chip_Ram.get(32L), GT_OreDictUnificator.get(OrePrefixes.wireFine, Materials.NiobiumTitanium, 24)}, tMat.getMolten(144L * tMultiplier), ItemList.Circuit_Biowarecomputer.get(1L), 400, 153600, true);
```
Comment (Dream-Master) [inline src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java:3465]: i would suggest tier up this one tier like smd recipes are cheaper than asmd recipe (tier wise)

**Review by Dream-Master — CHANGES_REQUESTED** [review]

File: src/main/java/com/dreammaster/bartworksHandler/BacteriaRegistry.java (hunk 1 above)
Comment (Dream-Master) [reply src/main/java/com/dreammaster/bartworksHandler/BacteriaRegistry.java:289]: ok yea use 1000 is very much. I would use 10 as values but yea

**Comment by GTNH-Colen** [comment]
This is effectively done now, only issue that needs resolving is the research conflict. My initial proposal of a new item with the xSMD in the corner which can be assembled with a circuit and xSMD stands. This makes logical sense as it is as if you are researching the circuit with the xSMD. If no one objects to this soon I will implement it because this change has dragged its feet for far too long due to way too much interference. Example image below. Also leaves open the door for even higher SMDs with similar research.
Screenshot 2022-02-09 at 04 15 31

**Comment by Dream-Master** [comment]
> This is effectively done now, only issue that needs resolving is the research conflict. My initial proposal of a new item with the xSMD in the corner which can be assembled with a circuit and xSMD stands. This makes logical sense as it is as if you are researching the circuit with the xSMD. If no one objects to this soon I will implement it because this change has dragged its feet for far too long due to way too much interference. Example image below. Also leaves open the door for even higher SMDs with similar research. Screenshot 2022-02-09 at 04 15 31

Lets make a concept first and not adding items we later not use in any recipes.

**Review by Dream-Master — APPROVED** [review]

**Pull request merged**